### ISSUE TITLE
[Feature]: Support for AWS Roles as Credentials for Backup Archive Upload to S3 Bucket

### ISSUE THREAD

**Issue by ame-appsmith** [appsmithorg/appsmith#25679]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Currently, the uploading of backup archives to S3 bucked is done with access keys.
Support for AWS Role as credential.

### Why should this be worked on?

Gives users more options. IAM Role is better from a security perspective compared to access keys.

**Comment by ronicthomas** [comment]
We need this as well - our environment doesn't allow long lived access keys to be created and all communication with AWS services must happen via IAM roles. As I understand, the code should remove all references to access keys and let the developers decide how they want the appsmith pods to communicate with AWS services

**Comment by sharat87** [comment]
Doesn't supporting this mean the entire node that's running Appsmith server will get read/write access to the S3 bucket? So anyone who knows the bucket name, with access to signup/login to that Appsmith can create a datasource to that S3 bucket and delete the backup archives? Or worse, plant a rogue backup archive in that bucket?

I don't think supporting this improves the security posture of Appsmith. Please correct if I'm misunderstanding this.

**Comment by ronicthomas** [comment]
Agreed that providing a role to the entire node can result in the above scenario. Propose modifying the backup script to accept an IAM role ARN as a parameter. The script would use AWS STS to assume this role, obtaining temporary credentials specifically for the backup operation.

Key benefits:
- Temporary, scoped credentials for backup
- No long-lived access keys
- Role can have minimal, precise permissions
- Credentials automatically expire after use

This approach ensures only the backup process can access the S3 bucket, with no broader node or application access.

**Comment by ronicthomas** [comment]
Any updates on this?
